### PR TITLE
Use h-assets package to serve compiled frontend assets

### PR DESCRIPTION
**Pull request by robertknight**
Replace the local code in `h/assets.py` for serving compiled JS and CSS assets listed with the h-assets package. The same change is being made for the lms project in https://github.com/hypothesis/lms/pull/2919. See that PR for the rationale and more details.

----

**Testing:**

- Make sure that the various views (eg. activity pages (`/search`) and admin pages) that reference compiled frontend assets still work
- Get the current URL of one of the assets (eg. from the browser's devtools) and try fetching it with the original query string and different query strings. You should get a 200 with the correct content if the cache buster is valid and a 404 if not

### CODE REVIEW

File: h/assets.py [hunk 1]
```diff
@@ -1,151 +1,19 @@
-import configparser
-import json
-import os
+import os.path
```
Comment (jon-betts) [inline h/assets.py:1]:
```suggestion
"""Add a view for getting static assets with cache busting."""

import os.path
```

Comment (robertknight) [reply h/assets.py:3]: Done, though with a slight revision to the wording.

File: h/assets.py [hunk 2]
```diff
@@ -1,151 +1,19 @@
...
-
-
-# Site assets
-assets_view = static_view("h:../build", cache_max_age=None, use_subpath=True)
-assets_view = _add_cors_header(assets_view)
+from h_assets import Environment, assets_view
 
 
 def includeme(config):
-    auto_reload = asbool(config.registry.settings.get("h.reload_assets", False))
```
[the remark this answers is not in the document]
Comment (robertknight) [reply h/assets.py:142]: I can't remember why this was originally removed but this setting is still needed. I re-added it.

File: h/assets.py [hunk 3]
```diff
@@ -1,151 +1,19 @@
...
-
-    config.add_view(route_name="assets", view=assets_view)
+    h_root = os.path.dirname(__file__)
 
     assets_env = Environment(
-        "/assets", "h/assets.ini", "build/manifest.json", auto_reload=auto_reload
+        assets_base_url="/assets",
+        bundle_config_path=f"{h_root}/assets.ini",
+        manifest_path=f"{h_root}/../build/manifest.json",
     )
```
[the remark this answers is not in the document]
Comment (robertknight) [reply h/assets.py]: I've made this change and it works, although the abstraction provided by `importlib_resources` is probably not terribly useful in this context given that `Environment` relies on a) the files existing on disk and b) the `build` directory located outside the root of the package. It makes the code consistent with other modules at least.

**Comment by robertknight** [comment]
I have updated this to re-add the auto-reload functionality which should not have been removed. I have also changed the code to use `importlib_resources`.

**Comment by robertknight** [comment]
Rebased on master due to conflicting changes with tests/h/assets_test.py (removed on this branch).

File: h/assets.py [hunk 4]
```diff
@@ -1,151 +1,19 @@
...
-        # or embedded.
-        #
-        # Some browsers enforce cross-origin restrictions on certain types of
-        # resources, eg. Firefox enforces same-domain policy for @font-face
-        # unless a CORS header is provided.
-        response = wrapped(context, request)
-        response.headers.extend({"Access-Control-Allow-Origin": "*"})
-        return response
-
-    return wrapper
```
[the remark this answers is not in the document]
Comment (robertknight) [reply h/assets.py:125]: We discussed this in Slack previously, but for the record, the CORS header added here is no longer required.

File: h/assets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jon-betts) [reply h/assets.py]: I've found a couple of useful things about this:

* Prevents rolling your own (although I find the new interface a bit weird honestly)
* Is guaranteed to be totally CWD agnostic which is good for running single tests (although this was already)
* May benefit from speed improvements as it uses Pythons module loading code which is apparently quite optimised for this

**Review by jon-betts — APPROVED** [review]
I bumped up the `h-assets` version to the latest for my local testing. But this is looking good I think.

**Pull request merged**